dspace-api/src/main/java/org/dspace/workflow/FlowStep.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.workflow;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.validation.constraints.NotNull;
+
+/**
+ * A workflow step.
+ */
+public class FlowStep {
```

Comment:
I'd appreciate it if we could improve the JavaDocs here.  This currently implies that a FlowStep is how you define a new Workflow Step.  But, that would then make it equivalent to `org.dspace.xmlworkflow.state.Step`  & we know that's not accurate.

It seems to me that this class is not actually defining a Workflow Step.  It's instead defining a mapping (is that the right word?) of Curation Task(s) to a Workflow Step defined in `workflow.xml`.  At least that's what I *think* is happening.

My point is that the lack of explanation here makes this very confusing.  I think the role of this FlowStep class is to map to a `<flowStep>` in `workflow-curation.xml`....but, it's unclear how that overlaps with the configurable workflow in `workflow.xml`...and whether the steps must share names or not.

Reply:
This reflects my own confusion when I began updating these classes.  You are correct:  these objects embody the relationships between workflow steps and curation tasks.  I hope that the commentary is clearer now.